src/Testcontainers/Builders/BuildConfiguration.cs

```diff
@@ -37,6 +37,25 @@ public static IEnumerable<T> Combine<T>(IEnumerable<T> next, IEnumerable<T> prev
       }
     }
 
+    /// <summary>
+    /// Combines all existing and new configuration changes while preserving the original order of insertion.
+    /// If there are no changes, the previous configurations are returned.
+    /// </summary>
+    /// <param name="next">Changed configuration.</param>
+    /// <param name="previous">Previous configuration.</param>
+    /// <typeparam name="T">Type of <see cref="IList{T}" />.</typeparam>
+    /// <returns>An updated configuration.</returns>
+    public static IList<T> Combine<T>(IList<T> next, IList<T> previous)
```

Comment:
True, but the reason I added this version of `Combine` is because there is technically nothing in the contract of the existing method that guarantees ordering of the combined enumerables, because `IEnumerable` does not guarantee preservation of order - `IList` does. So the fact that the existing `Combine` method preserves ordering is just a lucky implementation detail. At any time, this assumed contract could be broken either in the `Concat` method or in the implementation of `Combine`.

So I do not like the idea of treating the sequence of parameter modifiers as an `IEnumerable`, even if I could just convert the return value of `Combine` into a `List` - there is nothing in the contract that guarantees that my elements have not been shuffled around in the meantime.

Reply:
> because `IEnumerable` does not guarantee preservation of order

Yes - that's true, but the implementation does, and we choose the implementation. We can keep `IList` too, I just think it makes sense to stick to one implementation (and I prefer immutable).

Reply:
I agree with the immutable implementation. The `Combine` method should certainly not mutate the input enumerables. Depending on the other conversation, if we choose to remove `Combine<T>(IList<T> next, IList<T> previous)` and modify `Combine<T>(IEnumerable<T> next, IEnumerable<T> previous)` instead, I will add some unit tests to ensure that the implementation preserves ordering, despite the weaker contract.

Reply:
Let's keep `IList<T>`, see my comment below.